install/resources/upgrade50.php

```diff
@@ -29,7 +29,8 @@ function upgrade50_dbchanges()
 	echo "<p>Updating cache...</p>";
 
 	$cache->delete("banned");
-
+	
+	$db->update_query('users', array('pmfolders' => "0**$%%$1**$%%$2**$%%$3**$%%$4**"), "pmfolders NOT LIKE '0%'");
```

Comment:
A couple of things this doesn't address:
- users who have registered but not visited usercp/private.php `pmfolders=''`
- users who don't have the latest structure will loose their custom folders eg `pmfolders='1**$%%$2**$%%$3**$%%$4**$%%$5**Custom1Folder'`

Reply:
**It will cover all.**
- Users who have registered but not visited UCP must be having blank folder string in DB (which is not starting with 0), this will add new default structure for them.
 - Users who have custom folders must have visited `private.php` and a single call to that script has already corrected the structure which is starting with `0`, hence ignored by this upgrade line.
- Users who have visited will have old _default_ structure (starting with 1) with no custom folder. This will make that new _default_ structure.

Reply:
Yep sorry didn’t think through the empty one!

But for a user that has customised their folders by visiting private.php before the new structure will loose their customised folders. I tested it out earlier, reproduction steps:
1. Start with pre new structure copy of mybb
1. Customise folders as user (and then don’t load private.php or usercp.php)
1. Switch to this PR code
1. Run upgrade script
1. Go to private.php now and users folders have been lost

Reply:
I know, but I have no idea if we really need to care about the users created custom folders years back and never visited since then. If required at all; I will change the update script.
@euantorano ?

Reply:
I don't think that's very responsible. Anyone upgrading from 1.8.19 or older will have all custom folders lost!

Reply:
Something like this should do the trick
```suggestion
	$db->update_query('users', array('pmfolders' => "0**$%%$1**$%%$2**$%%$3**$%%$4**"), "pmfolders = ''");
	switch($db->type)
	{
		case "pgsql":
		case "sqlite":
			$update = "'0**$%%$' || pmfolders";
			break;
		default:
			$update = "CONCAT('0**$%%$', pmfolders)";
	}
	$db->write_query("UPDATE ".TABLE_PREFIX."users SET pmfolders=".$update." WHERE pmfolders NOT LIKE '0%'");
```